lib/chart/sparkline.ex

```diff
@@ -65,77 +117,143 @@ defmodule Contex.Sparkline do
+        area_class: "fill-blue-50"
+    )
+    |> Sparkline.draw()
+  ```
+
+  Example 3: Add a gradient
+
+  ```
+    extra_svg = \"""
+    <defs>
+      <linearGradient id="pretty-gradient" x1="0%" y1="0%" x2="0%" y2="100%">
+        <stop offset="0%" style="stop-color:#b794f4;stop-opacity:1" />
+        <stop offset="100%" style="stop-color:#f56565;stop-opacity:1" />
+      </linearGradient>
+    </defs>
+    \"""
+
+    Sparkline.new(data)
+    |> Sparkline.style(line_stroke: "url(#pretty-gradient)", extra_svg: extra_svg)
+    |> Sparkline.draw()
+  ```
+  """
+  def style(%Sparkline{} = sparkline, options \\ []) do
+    props =
+      @default_style
+      |> Keyword.merge(options)
+      |> Enum.into(%{})
+
+    Map.merge(sparkline, props)
   end
 
   @doc """
   Renders the sparkline to svg, including the svg wrapper, as a string or improper string list that
   is marked safe.
   """
-  def draw(%Sparkline{height: height, width: width, line_width: line_width} = sparkline) do
-    vb_width = sparkline.length + 1
-    vb_height = height - 2 * line_width
+  def draw(%Sparkline{} = chart) do
+    vb_width = chart.length + 1
+    vb_height = chart.height - 2 * chart.line_stroke_width
 
     scale =
       ContinuousLinearScale.new()
-      |> ContinuousLinearScale.domain(sparkline.data)
+      |> ContinuousLinearScale.domain(chart.data)
       |> Scale.set_range(vb_height, 0)
 
-    sparkline = %{sparkline | y_transform: Scale.domain_to_range_fn(scale)}
+    chart = %{chart | y_transform: Scale.domain_to_range_fn(scale)}
 
     output = ~s"""
-       <svg height="#{height}" width="#{width}" viewBox="0 0 #{vb_width} #{vb_height}" preserveAspectRatio="none" role="img">
-        <path d="#{get_closed_path(sparkline, vb_height)}" #{get_fill_style(sparkline)}></path>
-        <path d="#{get_path(sparkline)}" #{get_line_style(sparkline)}></path>
+       <svg height="#{chart.height}" width="#{chart.width}" viewBox="0 0 #{vb_width} #{vb_height}" preserveAspectRatio="none" role="img">
+        #{chart.extra_svg}
+        <path d="#{get_area_path(chart, vb_height)}" #{get_area_style(chart)}></path>
+        <path d="#{get_line_path(chart)}" #{get_line_style(chart)}></path>
       </svg>
     """
 
     {:safe, [output]}
   end
 
-  defp get_line_style(%Sparkline{line_colour: line_colour, line_width: line_width}) do
-    ~s|stroke="#{line_colour}" stroke-width="#{line_width}" fill="none" vector-effect="non-scaling-stroke"|
+  defp get_line_style(%Sparkline{
+         line_stroke: line_stroke,
+         line_stroke_width: line_stroke_width,
+         line_class: line_class,
+         line_fill: line_fill,
+         line_stroke_linecap: line_stroke_linecap,
+         line_stroke_linejoin: line_stroke_linejoin
+       }) do
+    ~s|stroke="#{line_stroke}" class="#{line_class}" stroke-width="#{line_stroke_width}" fill="#{line_fill}" stroke-linecap="#{line_stroke_linecap}" stroke-linejoin="#{line_stroke_linejoin}" vector-effect="non-scaling-stroke" |
```

Comment:
I haven't looked at the later commits yet, but it may be better to eliminate nil values. `Contex.SVG.opts_to_attrs` may help.